c/types.go

```diff
@@ -82,27 +100,66 @@ type NotifyStartFn = func(startError)
 // with the supervisor's ChildSpec.
 type ChildSpec struct {
 	name     string
+	tag      ChildTag
 	shutdown Shutdown
 	restart  Restart
 	start    func(context.Context, NotifyStartFn) error
 }
 
+// Tag returns the ChildTag of this ChildSpec
+func (cs ChildSpec) Tag() ChildTag {
+	return cs.tag
+}
+
+// IsWorker indicates if this child is a worker
+func (cs ChildSpec) IsWorker() bool {
+	return cs.tag == Worker
+}
+
 // Child is the runtime representation of an Spec
 type Child struct {
-	runtimeName string
-	spec        ChildSpec
-	cancel      func()
-	wait        func(Shutdown) error
+	runtimeName  string
+	spec         ChildSpec
+	restartCount uint32
+	cancel       func()
+	wait         func(Shutdown) error
+}
+
+// RuntimeName returns the name of this child (once started). It will have a
+// prefix with the supervisor name
+func (c Child) RuntimeName() string {
+	return c.runtimeName
+}
+
+// Name returns the name of the `ChildSpec` of this child
+func (c Child) Name() string {
+	return c.spec.name
+}
+
+// Spec returns the `ChildSpec` of this child
+func (c Child) Spec() ChildSpec {
+	return c.spec
+}
+
+// IsWorker indicates if this child is a worker
+func (c Child) IsWorker() bool {
+	return c.spec.IsWorker()
 }
 
 // ChildNotification reports when a child has terminated; if it terminated with
 // an error, it is set in the err field, otherwise, err will be nil.
 type ChildNotification struct {
+	name        string
```

Comment:
Why not also the tag (sorry, I am forgetting stuff, which may indicate the need to have a picture of the model 🤷‍♂ )

Reply:
Yep, that works, it doesn't hurt having it in the Events emitted